langchain/package.json

```diff
@@ -77,12 +78,16 @@
     "typedoc-plugin-missing-exports": "^1.0.0",
     "typescript": "^4.9.5"
   },
+  "peerDependencies": {
+    "pinecone-client": "^1.0.1"
+  },
```

Comment:
Let's add `peerDependenciesMeta` here too (https://docs.npmjs.com/cli/v9/configuring-npm/package-json#peerdependenciesmeta) so that langchain doesn't cause npm to emit warnings for users who aren't using pinecone.